src/zenml/hooks/hook_validators.py

```diff
@@ -67,10 +67,6 @@ def resolve_and_validate_hook(hook_func: "HookSpecification") -> str:
                     BaseException,
                     BaseParameters,
                     StepContext,
-                    # Have to do string version for TYPE_CHECKING
-                    "BaseException",
```

Comment:
But I added these just for the case.. Im very unsure about why we revert this now but if @schustmi and @strickvl are confident id say merge this PR in...

Reply:
That case doesn't really work in a nice way though: What if I pass my custom `BaseParameters` subclass as the type annotation there, it would already break down. I think we should just not allow these for now until we implement an actual solution to support string annotations in our steps/hooks.